tools/clang/test/HLSLFileCheck/d3dreflect/rdat_mintarget/sm66_atomicint64.hlsl

```diff
@@ -0,0 +1,96 @@
+// RUN: %dxilver 1.8 | %dxc -T lib_6_8 %s | %D3DReflect %s | %FileCheck %s -check-prefixes=RDAT,RDAT18
+// RUN: %dxilver 1.7 | %dxc -T lib_6_7 -validator-version 1.7 %s | %D3DReflect %s | FileCheck %s -check-prefixes=RDAT,RDAT17
+
+// Ensure min shader target incorporates optional features used
+
+// RDAT: FunctionTable[{{.*}}] = {
+
+// SM 6.6+
+
+///////////////////////////////////////////////////////////////////////////////
+// ShaderFeatureInfo_AtomicInt64OnTypedResource (0x400000) = 4194304
+
+// RDAT-LABEL: UnmangledName: "atomic_typed"
+// ShaderFeatureInfo_AtomicInt64OnTypedResource (0x400000) = 4194304
+// + ShaderFeatureInfo_Int64Ops (0x8000) = 0x408000 = 4227072
+// RDAT:   FeatureInfo1: 4227072
+// RDAT:   FeatureInfo2: 0
+// MinShaderTarget: (Library(6) << 16) + (SM 6.6 ((6 << 4) + 6)) = 0x60066 = 393318
+// RDAT: MinShaderTarget: 393318
+
+RWBuffer<uint64_t> RWBuf : register(u0, space0);
+
+[noinline] export
+void atomic_typed() {
+  uint64_t original;
+  InterlockedExchange(RWBuf[0], 12, original);
+}
+
+///////////////////////////////////////////////////////////////////////////////
+// ShaderFeatureInfo_AtomicInt64OnGroupShared (0x800000) = 8388608
+
+// RDAT-LABEL: UnmangledName: "atomic_groupshared"
+// ShaderFeatureInfo_AtomicInt64OnGroupShared (0x800000) = 8388608
+// + ShaderFeatureInfo_Int64Ops (0x8000) = 0x808000 = 8421376
+// RDAT:   FeatureInfo1: 8421376
+// RDAT:   FeatureInfo2: 0
+// MinShaderTarget: (Compute(5) << 16) + (SM 6.6 ((6 << 4) + 6)) = 0x50066 = 327782
+// RDAT18: MinShaderTarget: 327782
+// Old: 6.0
+// RDAT17: MinShaderTarget: 327776
+
+RWByteAddressBuffer BAB : register(u1, space0);
+groupshared int64_t gs;
+
+[shader("compute")]
+[numthreads(1,1,1)]
+void atomic_groupshared(uint tidx : SV_GroupIndex) {
+  if (tidx == 0)
+    gs = 0;
+  GroupMemoryBarrierWithGroupSync();
+  uint64_t original;
+  InterlockedExchange(gs, tidx, original);
+  BAB.Store(tidx * 4, original);
+}
+
+///////////////////////////////////////////////////////////////////////////////
+// ShaderFeatureInfo_AtomicInt64OnHeapResource (0x10000000) = 268435456
+
+// TBD: should it be set on bound heap resource?
```

Comment:
Oh yeah, we only set this if a dynamic heap resource is used with AtomicInt64.  I remember that was necessary, and was wondering why it wasn't set when you used a bound heap resource instead.  But we set a flag on the resource itself indicating this usage when it's a bound resource, and this flag makes it into PSV0 and RDAT resource flags, so I guess it wasn't necessary.  It was just a bit confusing because the feature can be used without incurring the feature flag requirement here.  I think it's by-design, but it made me go hmmm?

Reply:
I removed this comment.